cmd/allocate.go

```diff
@@ -69,9 +69,19 @@ var allocateCmd = &cobra.Command{
 			deployer = helper.GetDeployerByName(ctx, def.Deployer)
 		}
 
-		cluster, err := deployer.NewCluster(ctx, def)
-		if err != nil {
-			logger.Fatal("cluster deployment failed", zap.Error(err))
+		var cluster deployment.ClusterInfo
+		if def.Deployer == "Cloud" {
+			c, err := deployer.NewClusterOverRide(ctx, def)
```

Comment:
Since the `def` being passed here is actually the thing deciding whether there is a serverImage specified (and thus whether there is an Override needed), the decision on which path to take should be in the deployer, not at the top level (ie: having the DockerDeployer have NewClusterOverRide doesn't make sense).

Reply:
Yes, based on our previous discussion, I understood that we would be modifying the NewCluster function for the cloud to use the new path and remove the old one.

The updated NewCluster function now checks if a serverImage is provided. If so, it calls DeployNewCluster with the 'deploy' path; otherwise, it calls CreateNewCluster without the 'deploy' path.

Reply:
We can get release id from server-image, as each cloud images have this